ocs_ci/ocs/node.py

```diff
@@ -1464,3 +1456,65 @@ def wait_for_new_osd_node(old_osd_node_names, timeout=180):
     except TimeoutExpiredError:
         log.warning(f"New osd node didn't appear after {timeout} seconds")
         return None
+
+
+def add_new_node_and_label_upi_lso(
+    node_type,
+    num_nodes,
+    mark_for_ocs_label=True,
+    node_conf=None,
+    attach_disk=True,
+    disk_size=None,
+):
+    """
+    Add a new node for aws/vmware upi lso platform and label it
+
+    Args:
+        node_type (str): Type of node, RHEL or RHCOS
+        num_nodes (int): number of nodes to add
+        mark_for_ocs_label (bool): True if label the new node
+        node_conf (dict): The node configurations.
+        attach_disk (bool): True, if attach a new disk to the new node. False otherwise
+        disk_size (int): The size of the new disk to attach. If not specified,
+            the disk size will be equal to the size of the previous disk.
+
+    Returns:
+        list: new spun node names
+
+    """
+    new_node_names = add_new_node_and_label_upi(
+        node_type, num_nodes, mark_for_ocs_label, node_conf
+    )
+    new_node_objs = get_node_objs(node_names=new_node_names)
+
+    # Check if we need to attach disk to the new nodes
+    if attach_disk:
+        for node_obj in new_node_objs:
+            log.info(
+                f"Create a new disk with size {disk_size}, and attach to node {node_obj.name}"
+            )
+            add_disk_to_upi_node(node_obj, disk_size)
+
+    return new_node_names
+
+
+def add_disk_to_upi_node(node_obj, disk_size=None):
+    """
+    Add new disk to lso upi node
+
+    Args:
+        node_obj (ocs_ci.ocs.resources.ocs.OCS): The node object
+        disk_size (int): The size of the new disk to attach. If not specified,
+            the disk size will be equal to the size of the previous disk.
+
+    """
+    from ocs_ci.ocs.platform_nodes import PlatformNodesFactory
```

Comment:
move line 1511 to the beginning of the file

Reply:
I put it here and not at the top of the file because it causes a circular loop error. 
You can also see the same import at this line https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/node.py#L357.